### DISCUSSION TITLE
Login Issues

### DISCUSSION THREAD

**Discussion by benpayne** [inventree/InvenTree#6751] [Q&A] [answered]
I have a deployment using the docker container deployed to kubernetes.  I've had this site running for several months.  But I had it setup to deploy "stable", so it was periodically updating and would require DB migration.  I'd jump on and do the migration when it showed up and all was good.  But recently I did the migration, all looked good, but now I can't login.  I'm able to access it from my main browser, I think because of a cookie.  But on another browser it fails to login.  I've been using google for SSO, but I've disabled this for now and it still fails. It just gives and "authentication error" always. Even on my main browser where I can access the site, I get auth failures if I try and create and change anything.  I'm struggling with how to debug this.  nothing obvious is showing on the server logs.  Any ideas on how to get something to indicate the issue would be greatly appreciated.  I've not fixed this deployment at 0.14.3, all migrations are done and the server and workers are updated to 0.14.3.

**Comment by SchrodingersGat** [comment]
@benpayne after updating to version `0.14.x` you will need to review your "site access settings":

https://docs.inventree.org/en/latest/start/config/#server-access

In particular:

- Are you running behind a proxy?
- Are you trying to access the server from a different host?

**Reply by benpayne** [reply]
I have an nginx proxy setup that also server the media and static files.  I thought maybe the static files were out of date, so I made sure they were updated.  Yes this is running on a cloud provider so accessing it remotely.  This is my nginx proxy config.  

```
    server {
        listen 443 ssl;
        server_name inventree.domain.com;

        ssl_certificate /var/www/live/inventree.domain.com/fullchain.pem;
        ssl_certificate_key /var/www/live/inventree.domain.com/privkey.pem;

        real_ip_header proxy_protocol;

        location / {

            proxy_set_header      Host              $http_host;
            proxy_set_header      X-Forwarded-By    $server_addr:$server_port;
            proxy_set_header      X-Forwarded-For   $remote_addr;
            proxy_set_header      X-Forwarded-Proto $scheme;
            proxy_set_header      X-Real-IP         $remote_addr;
            proxy_set_header      CLIENT_IP         $remote_addr;

            proxy_pass_request_headers on;

            proxy_redirect off;

            client_max_body_size 100M;

            proxy_buffering off;
            proxy_request_buffering off;

            # Do not touch this unless you have a specific reason - this and the docker-compose need to match
            proxy_pass http://inventree-server:8000;
        }

        # Redirect any requests for static files
        location /static/ {
            alias /var/www/static/;
            autoindex on;

            # Caching settings
            expires 30d;
            add_header Pragma public;
            add_header Cache-Control "public";
        }

        # Redirect any requests for media files
        location /media/ {
            alias /var/www/media/;

            # Media files require user authentication
            auth_request /auth;

            # Content header to force download
            add_header Content-disposition "attachment";
        }

        # Use the 'user' API endpoint for auth
        location /auth {
            internal;

            proxy_pass http://inventree-server:8000/auth/;

            proxy_pass_request_body off;
            proxy_set_header Content-Length "";
            proxy_set_header X-Original-URI $request_uri;
        }

        # Redirect requests for let's encrypt authentication
        location /.well-known/acme-challenge/ {
            root /var/www/letsencrypt/;
        }
    }
```

**Reply by benpayne** [reply]
I though the static files might be an issue since I'm getting an error on the proxy that this file is not found `/var/www/static/script/chartjs-adapter-moment.min.js.map`.  But I don't think that's blocking the login.  But these are the errors:

```
0.128.0.56 - - [19/Mar/2024:05:09:04 +0000] "POST /accounts/login/ HTTP/1.1" 403 6946 "https://inventree.brsoft.io/accounts/login/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36" "-"
2024/03/19 05:09:05 [error] 21#21: *265 open() "/var/www/static/script/chartjs-adapter-moment.min.js.map" failed (2: No such file or directory), client: 10.128.15.206, server: inventree.brsoft.io, request: "GET /static/script/chartjs-adapter-moment.min.js.map HTTP/1.1", host: "inventree.brsoft.io"
10.128.15.206 - - [19/Mar/2024:05:09:05 +0000] "GET /static/script/chartjs-adapter-moment.min.js.map HTTP/1.1" 404 555 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36" "-"
```

**Reply by benpayne** [reply]
@SchrodingersGat when you say "site access settings", can you be more specific to what you're referring to.  What there a change in 0.14.x that I specifically need to pay attention to?  If you can point to the Issue or commit that would be helpful.

**Reply by benpayne** [reply]
Ok just notices this in the server:
```
 Forbidden (Origin checking failed - https://inventree.brsoft.io does not match any trusted origins.): /accounts/login/   
```
So I guess your saying there's an issue with CORS and the something in 0.14.x changed with regard to this...

**Reply by wolflu05** [reply]
That error is fine, that exists always, because somewhere is a source map recerenced that does not exist, but as we are already in rewriting the frontend noone has investigated this further.

And regarding the access options, have you had a look at https://docs.inventree.org/en/stable/start/config/#server-access ? How have you configured `INVENTREE_SITE_URL`, and how do you access Inventree?

**Reply by benpayne** [reply, marked as answer]
So setting INVENTREE_SITE_URL resolved this issue.  I was a bit confused since "base_url" was set in the config already.  Not sure what the difference between "base_url" and "site_url" is, both are now set to the same value and it works.

**Reply by SchrodingersGat** [reply]
@benpayne `base_url` was not actually doing anything - the variable was mis-labelled in an older version and may have just "stuck around" in your configuration.. Apologies!

**Reply by benpayne** [reply]
Ok good to know.  

Maybe it should be removed from the documentation then too?  It's still in the "Basic Options" section just above the link mentioned earlier.

**Reply by SchrodingersGat** [reply]
@benpayne thanks for letting me know - that's an error in the docs. I'll get that fixed ASAP

**Reply by pyhtoncoder0** [reply]
I am new to InvenTree and over all a novas at docker. However I too am running a production system under :stable. After doing my update last night I am having this exact issue. However I'm not sure where to specify INVENTREE_SITE_URL. Do I just add this under "environment"  in the docker-compose/.env files?

hosted on docker
Id like to remove all proxy stuff tbh. But I don't know how. I use other services to tunnel securely into the network. Any help just removing all this would be appreciated also. However mean time i've just left it as I don't know how to remove properly. I attempted a bare metal installation without proxy but was never able to even access the service locally. I apologize, I know helping with these issues is tedious. I'm a new IT admin of sorts and really struggling right now. I have a full team of people down my case to fix this and I don't really know what I'm doing.

**Reply by pyhtoncoder0** [reply]
After messing with it a bit I can now access locally. I changed the INVENTREE_SITE_URL in the settings page. I've tried a local address as well as the true domain used. Nether allows logging in thu public channels. But I have full functionality back at least locally. At this location there is no public IP available to this application. I use cloudflared to reach these services publicly. I have a lot of work built out on cloudflare for user security. This is why I would like to just remove all proxy at the application level. I have other proxies running and the cloudflare to get into the network. They usually play nice... but not today. I also would like to note in the past I had issues getting the service public when first deploying. I had to really mess with the address and find something that worked. However it never killed access for the poor configuration, It just warned me that a config may cause issues. I of course did investigate and was able to remove the error by fulfilling what it asked. However I don't remember now what the issue was in the end. Maybe this information helps. Maybe not...

**Reply by pyhtoncoder0** [reply]
So... I am so stupid. I figured out where the value is placed. I even had it set already! INVENTREE_SITE_URL is found in the .env file for your install. I have mine in /apps/inventree/.env along with the docker-compose.yml file... also at /home/inventree/docker-compose.yml. I'm not sure what happened but the line got commented out. All I had to is un-comment the line, restart the server and all is well again. Sorry for the freak out. I was struggling on all fronts. Won't even lie.

**Reply by SchrodingersGat** [reply]
If there are any areas where you think that the documentation could be improved around setup, please let us know!

**Reply by Kexe-Dev** [reply]
Hello, I have a similar problem when attempting to login to my http://inventree.localhost/

When trying to login either with default user or admin user account, I get `Forbidden (CSRF cookie not set.): /accounts/login/` error in my docker inventree-server console log. I am pretty confused with all django configuration and I dont know how to fix this problem. I dont know if I gave you relevant info but just ask and Ill try to give you more.

**Reply by SchrodingersGat** [reply]
@bebe-Kexe are you installing via the docker setup guide? https://docs.inventree.org/en/latest/start/docker_install/

**Reply by jbrryhooves** [reply]
I am getting the same error as [bebe-Kexe](https://github.com/bebe-Kexe) with a remote install. 
I followed the docker install guide, changed the bare minimum from the examples (`INVENTREE_SITE_URL` ) and get the CSRF cookie error. 
The new platform UI works, but the old (current?) one doesn't. 

.env snippet
```
# Site URL - update this to match your host
INVENTREE_SITE_URL='https://stock.mycompany.com'
INVENTREE_TRUSTED_ORIGINS=["https://stock.mycompany.com","http://stock.mycompany.com","http://inventree-server:8000"]

COMPOSE_PROJECT_NAME=inventree
```

**Reply by SchrodingersGat** [reply]
@jbrryhooves what version are you on? We just released [version 0.6.7](https://github.com/inventree/InvenTree/releases/tag/0.16.7) which addresses some of these login issues

**Reply by jbrryhooves** [reply]
ok thanks - I pulled "stable" a few days ago. Just updated to the latest "stable" version (0.16.7) and it didn't work out the box, but I set `INVENTREE_COOKIE_SAMESITE` to `false` in `.env` and that worked. 
Thanks for the help

**Reply by Kexe-Dev** [reply]
> @bebe-Kexe are you installing via the docker setup guide? https://docs.inventree.org/en/latest/start/docker_install/



Yes, I am following that setup guide.

I CD to my working directory where my base files are located and just copy-paste those commands that are in that documentation to my cmd and that does all the work. All I do is set the URL, user credentials and admin credentials to the .env file. The site runs but I can't login. If I had to configure djangos config file, I would be pretty lost.

**Reply by Kexe-Dev** [reply]
Ive noticed that on inventree-server docker start, `Not all required settings for adding a user on startup are present:
2024-10-29 17:16:45 INVENTREE_ADMIN_USER, INVENTREE_ADMIN_EMAIL, INVENTREE_ADMIN_PASSWORD` argument gets displayed even though my .env has `INVENTREE_ADMIN_USER` specified and the line isnt commented out... 

Also whats the difference between .env file an config.yaml? Like its the same as .env but different at the same time? I am really confused.

**Reply by Kexe-Dev** [reply]
Got it working!
Thanks everyone for help!

**Reply by turist79** [reply]
> ok thanks - I pulled "stable" a few days ago. Just updated to the latest "stable" version (0.16.7) and it didn't work out the box, but I set `INVENTREE_COOKIE_SAMESITE` to `false` in `.env` and that worked. Thanks for the help

man... you saved me!

for those, who will have issues logging in:
if you are getting "You have been logged out from InvenTree." immediately after providing correct login and password - check server log (in my case - logs from inventree_server docker container), if there is **Forbidden (CSRF cookie not set.): /accounts/login/** error, add INVENTREE_COOKIE_SAMESITE variable and set it to "false".

PS: i have no idea what it does and why i was able to log in for the first time. I had situation, when i was able to login only from my notebook which i used first, but later not from any other device like android phone for example.
